### PR TITLE
Fix sharing imap connections between different clients

### PR DESCRIPTION
**Pull request by blaxter**
Two main fixes:

1. Disable current user context when an operation has ended.
2. Enable user context before creating sogo objects.

Without this there were situations where you could use an imap connection from another user because some objects were created with a WOContext with an active user different of the current one (but from a previous request).

Suggested messages for NEWS:
* Fixed several sync issues on environments with multiple users
* Folders from other users will no longer appear on your Outlook

### CODE REVIEW

File: OpenChange/MAPIStoreContext.m [hunk 1]
```diff
@@ -114,7 +113,6 @@ + (struct mapistore_contexts_list *) listAllContextsForUser: (NSString *)  userN
 
   userContext = [MAPIStoreUserContext userContextWithUsername: userName
                                                andTDBIndexing: indexing];
-  [userContext activateWithUser: [userContext sogoUser]];
```
Comment (sixstone-qq) [inline OpenChange/MAPIStoreContext.m]: This change makes the `userContext` getter useless according to the compiler. See:
```
 Compiling file MAPIStoreContext.m ...
MAPIStoreContext.m: In function ‘+[MAPIStoreContext listAllContextsForUser:withIndexing:inMemCtx:]’:
MAPIStoreContext.m:110:25: warning: variable ‘userContext’ set but not used [-Wunused-but-set-variable]
   MAPIStoreUserContext *userContext; 
```

**Comment by sixstone-qq** [comment]
Apart from this minor thing, I wonder if any missing section to activate the user is done, I hope it is not.

No regressions found while testing and there is no reproducible way to provoke the problem so :+1:

File: OpenChange/MAPIStoreContext.m (hunk 1 above)
Comment (blaxter) [inline OpenChange/MAPIStoreContext.m]: yep, but the userContext needs to be created

Comment (sixstone-qq) [inline OpenChange/MAPIStoreContext.m]: Could we avoid it somehow? :) No warnings in the output would be great.

**Comment by papajulio** [comment]
:+1: To the merge as the tests are passing correctly except the regression in Zinc tests

**Pull request merged**